app/api/data.py

```diff
@@ -5,43 +5,96 @@
 from app.models.data import *
 from app import db
 
+##############################################################################################
+######################################   Health check      ###################################
+##############################################################################################
 
-@api.route('/test/', methods=['GET'])
+
+@api.route('/test', methods=['GET'])
 def get_data():
     current_app.logger.info('Retrieving all data: placeholder')
     current_app.logger.debug('This is a debug log')
     return jsonify({'test_data_key': 'test_data_value'})
 
 
-@api.route('/data/batch/', methods=['GET'])
+##############################################################################################
+######################################   Batches      ########################################
+##############################################################################################
+
+
+@api.route('/batches', methods=['GET'])
 def get_batches():
     current_app.logger.info('Retrieving all batches')
     batches = Batch.query.all()
+    # for each batch, attach its coreData rows
+    
     return jsonify({
         'batches': [batch.to_dict() for batch in batches]
     })
 
-@api.route('/data/state/', methods=['GET'])
+
+##############################################################################################
+######################################   States      #########################################
+##############################################################################################
+
+@api.route('/states', methods=['GET'])
 def get_states():
     current_app.logger.info('Retrieving all states')
     states = State.query.all()
     return jsonify({
         'states': [state.to_dict() for state in states]
     })
 
-@api.route('/data/', methods=['GET'])
-def get_core_data():
-    current_app.logger.info('Retrieving all states')
-    data = CoreData.query.join(Batch).join(State).all()
-    return jsonify({
-        'data': [x.to_dict() for x in data]
-    }) 
+##############################################################################################
+######################################   Core data      ######################################
+##############################################################################################
+
 
-@api.route('/data/batch/', methods=['POST'])
-def post_data():
+@api.route('/batches', methods=['POST'])
+def post_core_data():
+    """
+    Workhorse POST method for core data
+
+    Requirements: 
+    """
     current_app.logger.info('Got a post request!')
-    payload = request.json
-    if not payload:
-        abort(400)
-    current_app.logger.info(payload)
-    return jsonify({'payload': payload})
+    payload = request.json  # this is a dict
+
+    # test the input data
+    assert 'context' in payload, "Payload requires 'context' field"
```

Comment:
This can be a follow-up issue to address integrity checking more generally rather than something to change right now, but we should return a status 400 (or maybe 422 if we want to be pedantic but whatever) with an array of validation errors when assertions like these fail. I'm fine with the assertions for now though.